clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -290,7 +295,13 @@ PointerVariableConstraint::PointerVariableConstraint(
       VarCreated = true;
       assert(CAtom != nullptr && "Unable to find the type "
                                  "of the checked pointer.");
-      Vars.push_back(CAtom);
+      if (VarAtomForChecked) {
```

Comment:
What is `VarAtomForChecked` doing?

Reply:
Is this the idea that when you re-run you use a fresh variable for the internal CV, so you end up re-deriving its solution?